### DISCUSSION TITLE
how to add a new hard device?

### DISCUSSION THREAD

**Discussion by cnbattle** [rook/rook#9820] [Q&A] [answered]
k8s  rook now info 

3 worker node, Each node has 1 T hard disk, install  rook ceph  cluster make up 3t raw capacity .

i have added a 1t hard disk to a node,

CephCluster yaml storage setting:
```
  storage: 
    useAllNodes: true
    useAllDevices: true
```
operator  yaml setting:
```
ROOK_ENABLE_DISCOVERY_DAEMON: "true"
```

i run ` kubectl rollout restart deploy rook-ceph-operator -n rook-ceph`, not in effect.

but now ,  how do I add this hard disk to the cluster?

**Comment by BlaineEXE** [comment]
You should check the results of the rook-ceph-osd-prepare pods. It seems likely that the disks added might have an old partition table or data on them. If so, Rook will not erase that data in order to proceed with installation. You must wipe data from those disks first.

**Comment by travisn** [comment]
The [troubleshooting guide](https://rook.github.io/docs/rook/v1.8/ceph-common-issues.html#osd-pods-are-not-created-on-my-devices) may help as well.

**Comment by cnbattle** [comment, marked as answer]
i setting `cluster.yaml` ` monitoring.enabled: true` ,but i donot insrtall prometheus
```
  monitoring:
    enabled: true
```
so make a mistake, did not run a disk operation to automatic process.

 i change to `false`, is ok

thanks everyone.
